[X86] Update BMI, BMI2 and TBM constexpr tests

bursow opened the pull request:
This PR updates the constexpr tests for the BMI, BMI2, and TBM intrinsics. The existing `char` array-based tests are replaced with the `TEST_CONSTEXPR` helper macro to improve readability and align with modern testing practices in the project.

Files modified:
- clang/test/CodeGen/X86/bmi-builtins.c
- clang/test/CodeGen/X86/bmi2-builtins.c
- clang/test/CodeGen/X86/tbm-builtins.c

Commit:
[X86] Update BMI, BMI2 and TBM constexpr tests

Update constexpr tests in these files to use the TEST_CONSTEXPR helper macro for clarity and consistency.

Commit:
Fix: Use double quotes for builtin_test_helpers.h include

Commit:
Merge branch 'main' into add-bmi-tbm-tests

Commit:
Fix: builtin_test_helpers.h

Commit:
Fix: C++11 compatibility issues in builtin test helpers

Commit:
Merge branch 'main' into add-bmi-tbm-tests

bursow commented:
Checked C++14-specific variable declarations within constexpr functions, which were causing compilation errors in the C++11 environment. refactored the code to be C++11 compliant, which resolved the issue also added BMI, BMI2 and TBM constexpr tests. @RKSimon

Commit:
Merge branch 'llvm:main' into add-bmi-tbm-tests

bursow commented on clang/test/CodeGen/X86/builtin_test_helpers.h:
```diff
@@ -4,252 +4,220 @@
 #if defined(__cplusplus) && (__cplusplus >= 201103L)
 
+
+#define TEST_CONSTEXPR(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
```
Hi, I'm a bit confused. The tests pass because the C++11 compatibility issue in builtin_test_helpers.h was resolved. If I revert those changes, the tests fail again. Is there another way I should be approaching this?

bursow commented on clang/test/CodeGen/X86/bmi-builtins.c:
```diff
@@ -237,85 +238,85 @@ unsigned long long test_blsr_u64(unsigned long long __X) {
 #if defined(__cplusplus) && (__cplusplus >= 201103L)
-char andnu32[__andn_u32(0x01234567, 0xFECDBA98) == (~0x01234567 & 0xFECDBA98) ? 1 : -1];
-char andn2u32[_andn_u32(0x01234567, 0xFECDBA98) == (~0x01234567 & 0xFECDBA98) ? 1 : -1];
+TEST_CONSTEXPR(__andn_u32(0x01234567, 0xFECDBA98) == (~0x01234567 & 0xFECDBA98));
```
Okey

Commit:
Refactor: Move constexpr tests next to their respective functions

Commit:
Refactor: Move constexpr tests next to their respective functions

Commit:
tests(X86): fix _mm_tzcnt_64 typo and close conditionals properly

RKSimon commented on clang/test/CodeGen/X86/bmi-builtins.c:
```diff
@@ -22,49 +22,89 @@ unsigned short test_tzcnt_u16(unsigned short __X) {
 // TZCNT: i16 @llvm.cttz.i16(i16 %{{.*}}, i1 false)
   return _tzcnt_u16(__X);
 }
+#if defined(__cplusplus) && (__cplusplus >= 201103L)
```
these aren't necessary

RKSimon commented on clang/test/CodeGen/X86/builtin_test_helpers.h:
```diff
@@ -296,4 +296,4 @@ constexpr bool match_v64qu(__m512i _v, unsigned char __e00, unsigned char __e01,
 #define TEST_CONSTEXPR(...)
 
-#endif
+#endif
```
no newline

RKSimon commented on clang/test/CodeGen/X86/bmi2-builtins.c:
```diff
@@ -54,22 +55,22 @@ unsigned long long test_mulx_u64(unsigned long long __X, unsigned long long __Y,
-char pext32_1[_pext_u32(0x89ABCDEF,  0x000000F0) == 0x0000000E ? 1 : -1];
-char pext32_2[_pext_u32(0x89ABCDEF,  0xF00000F0) == 0x0000008E ? 1 : -1];
-char pext32_3[_pext_u32(0x89ABCDEF,  0xFFFFFFFF) == 0x89ABCDEF ? 1 : -1];
+TEST_CONSTEXPR(_bzhi_u32(0x89ABCDEF,   0) == 0x00000000);
```
fix this

Commit:
[clang][test] Refactor bmi2-builtins.c constexpr tests

Move TEST_CONSTEXPR checks directly under each function
instead of grouping them.

Commit:
bmi-builtins.c
